### PR TITLE
DAT-4474 | Tests for Load More functionality on mercury search result page

### PR DESCRIPTION
**Pull request by rybmat**
https://wikia-inc.atlassian.net/browse/DAT-4474
https://wikia-inc.atlassian.net/browse/DAT-4150

@Wikia/west-wing 

propositions of any additional test cases are welcome

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/elements/mercury/components/Search.java [hunk 1]
```diff
@@ -13,6 +13,8 @@
 import org.openqa.selenium.interactions.Actions;
 import org.openqa.selenium.support.FindBy;
 
+import java.util.List;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/Search.java]: MINOR Remove this unused import 'java.util.List'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/elements/mercury/pages/SearchResultsPage.java [hunk 2]
```diff
@@ -76,14 +85,45 @@ public boolean isNoResultsPagePresent() {
...
+
+  public int getResultCardsNumber() {
+    waitForResultsLoaded();
+    return resultCards.size();
+  }
+
+  private boolean isElementVisible(WebElement element) {
+    try {
+      return element.isDisplayed();
+    } catch (NoSuchElementException e) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/pages/SearchResultsPage.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/SearchTests.java [hunk 3]
```diff
@@ -180,4 +185,55 @@ public void mercury_search_redirectToNewResultsPageFromNoResults() {
...
+        new SearchResultsPage()
+            .openForQuery(SINGLE_RESULT_SEARCH_PHRASE);
+
+    Assertion.assertTrue(resultsPage.getResultCardsNumber() < SEARCH_RESULTS_DEFAULT_NUMBER);
+    Assertion.assertFalse(resultsPage.isLoadMoreButtonVisible());
+  }
+
+  @InBrowser(emulator = Emulator.GOOGLE_NEXUS_5)
+  @Test(groups = "mercury_search_loadMoreResultsOnSearchResultsPageNotVisible")
+  public void mercury_search_emptySearchPhrase() {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/SearchTests.java]: MAJOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00100)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/SearchTests.java [hunk 4]
```diff
@@ -180,4 +185,55 @@ public void mercury_search_redirectToNewResultsPageFromNoResults() {
...
+    SearchResultsPage resultsPage =
+        new SearchResultsPage()
+            .openForQuery(MULTIPLE_RESULTS_SEARCH_PHRASE);
+
+    Assertion.assertEquals(resultsPage.getResultCardsNumber(), SEARCH_RESULTS_DEFAULT_NUMBER);
+  }
+
+  @InBrowser(emulator = Emulator.GOOGLE_NEXUS_5)
+  @Test(groups = "mercury_search_loadingMoreResultsOnSearchResultsPage")
+  public void mercury_search_loadingMoreResultsOnSearchResultsPage() {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/SearchTests.java]: MAJOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00100)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/SearchTests.java [hunk 5]
```diff
@@ -180,4 +185,55 @@ public void mercury_search_redirectToNewResultsPageFromNoResults() {
...
+    resultsPage.clickLoadMoreButton();
+    int moreResultsLoaded = resultsPage.getResultCardsNumber() - defaultCardNumber;
+
+    Assertion.assertEquals(moreResultsLoaded, SEARCH_RESULTS_DEFAULT_NUMBER);
+    Assertion.assertTrue(resultsPage.isLoadMoreButtonVisible());
+  }
+
+  @InBrowser(emulator = Emulator.GOOGLE_NEXUS_5)
+  @Test(groups = "mercury_search_loadMoreResultsOnSearchResultsPageNotVisible")
+  public void mercury_search_loadMoreResultsOnSearchResultsPageNotVisible() {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/SearchTests.java]: MAJOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00100)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/SearchTests.java [hunk 6]
```diff
@@ -180,4 +185,55 @@ public void mercury_search_redirectToNewResultsPageFromNoResults() {
     Assertion.assertFalse(searchResults.isNoResultsPagePresent());
     Assertion.assertTrue(searchResults.areResultsPresent());
   }
+
+  @InBrowser(emulator = Emulator.GOOGLE_NEXUS_5)
+  @Test(groups = "mercury_search_defaultResultsNumberOnSearchResultsPage")
+  public void mercury_search_defaultResultsNumberOnSearchResultsPage() {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/SearchTests.java]: MAJOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00100)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 6 issues:
* MAJOR 5 major
* MINOR 1 minor

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/elements/mercury/pages/SearchResultsPage.java [hunk 7]
```diff
@@ -23,10 +25,17 @@
   @FindBy(css = ".search-error-not-found__action")
   private WebElement tryAnotherSearchLink;
 
+  @FindBy(css = ".wikia-button")
```
Comment (SebastianMarzjan) [inline src/test/java/com/wikia/webdriver/elements/mercury/pages/SearchResultsPage.java]: is this selector specific enough?

Comment (dianafa) [inline src/test/java/com/wikia/webdriver/elements/mercury/pages/SearchResultsPage.java]: more specific would be `.search-results__load-more-wrapper .wikia-button`

Comment (rybmat) [inline src/test/java/com/wikia/webdriver/elements/mercury/pages/SearchResultsPage.java]: good point, I didn't notice that

File: src/test/java/com/wikia/webdriver/elements/mercury/pages/SearchResultsPage.java [hunk 8]
```diff
@@ -23,10 +25,17 @@
...
+  private WebElement loadMoreButton;
+
+  @FindBy(css = ".wikia-card")
```
Comment (SebastianMarzjan) [inline src/test/java/com/wikia/webdriver/elements/mercury/pages/SearchResultsPage.java]: Same here :)

**Comment by SebastianMarzjan** [comment]
Only the selector specifity from my end, rest LGTM

File: src/test/java/com/wikia/webdriver/elements/mercury/components/Search.java (hunk 1 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/Search.java]: MINOR Remove this unused import 'java.util.List'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/elements/mercury/pages/SearchResultsPage.java (hunk 2 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/pages/SearchResultsPage.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/SearchTests.java (hunk 3 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/SearchTests.java]: MAJOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00100)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/SearchTests.java (hunk 4 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/SearchTests.java]: MAJOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00100)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/SearchTests.java (hunk 5 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/SearchTests.java]: MAJOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00100)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/SearchTests.java (hunk 6 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/SearchTests.java]: MAJOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00100)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 6 issues:
* MAJOR 5 major
* MINOR 1 minor

Watch the comments in this conversation to review them.

**Comment by WikiaTech** [comment]
SonarQube analysis reported 5 issues:
* MAJOR 5 major

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/elements/mercury/pages/SearchResultsPage.java (hunk 2 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/pages/SearchResultsPage.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/SearchTests.java [hunk 9]
```diff
@@ -180,4 +184,55 @@ public void mercury_search_redirectToNewResultsPageFromNoResults() {
...
+        new SearchResultsPage()
+            .openForQuery(SINGLE_RESULT_SEARCH_PHRASE);
+
+    Assertion.assertTrue(resultsPage.getResultCardsNumber() < SEARCH_RESULTS_DEFAULT_NUMBER);
+    Assertion.assertFalse(resultsPage.isLoadMoreButtonVisible());
+  }
+
+  @InBrowser(emulator = Emulator.GOOGLE_NEXUS_5)
+  @Test(groups = "mercury_search_loadMoreResultsOnSearchResultsPageNotVisible")
+  public void mercury_search_emptySearchPhrase() {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/SearchTests.java]: MAJOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00100)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/SearchTests.java [hunk 10]
```diff
@@ -180,4 +184,55 @@ public void mercury_search_redirectToNewResultsPageFromNoResults() {
...
+    SearchResultsPage resultsPage =
+        new SearchResultsPage()
+            .openForQuery(MULTIPLE_RESULTS_SEARCH_PHRASE);
+
+    Assertion.assertEquals(resultsPage.getResultCardsNumber(), SEARCH_RESULTS_DEFAULT_NUMBER);
+  }
+
+  @InBrowser(emulator = Emulator.GOOGLE_NEXUS_5)
+  @Test(groups = "mercury_search_loadingMoreResultsOnSearchResultsPage")
+  public void mercury_search_loadingMoreResultsOnSearchResultsPage() {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/SearchTests.java]: MAJOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00100)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/SearchTests.java [hunk 11]
```diff
@@ -180,4 +184,55 @@ public void mercury_search_redirectToNewResultsPageFromNoResults() {
...
+    resultsPage.clickLoadMoreButton();
+    int moreResultsLoaded = resultsPage.getResultCardsNumber() - defaultCardNumber;
+
+    Assertion.assertEquals(moreResultsLoaded, SEARCH_RESULTS_DEFAULT_NUMBER);
+    Assertion.assertTrue(resultsPage.isLoadMoreButtonVisible());
+  }
+
+  @InBrowser(emulator = Emulator.GOOGLE_NEXUS_5)
+  @Test(groups = "mercury_search_loadMoreResultsOnSearchResultsPageNotVisible")
+  public void mercury_search_loadMoreResultsOnSearchResultsPageNotVisible() {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/SearchTests.java]: MAJOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00100)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/SearchTests.java [hunk 12]
```diff
@@ -180,4 +184,55 @@ public void mercury_search_redirectToNewResultsPageFromNoResults() {
     Assertion.assertFalse(searchResults.isNoResultsPagePresent());
     Assertion.assertTrue(searchResults.areResultsPresent());
   }
+
+  @InBrowser(emulator = Emulator.GOOGLE_NEXUS_5)
+  @Test(groups = "mercury_search_defaultResultsNumberOnSearchResultsPage")
+  public void mercury_search_defaultResultsNumberOnSearchResultsPage() {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/SearchTests.java]: MAJOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00100)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 5 issues:
* MAJOR 5 major

Watch the comments in this conversation to review them.

**Pull request merged**